cli/tests/pcluster/validators/test_secret_validators.py

```diff
@@ -0,0 +1,157 @@
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from pcluster.aws.common import AWSClientError
+from pcluster.validators.common import FailureLevel
+from pcluster.validators.secret_validators import ArnServiceAndResourceValidator, MungeKeySecretSizeAndBase64Validator
+from tests.pcluster.validators.utils import assert_failure_level, assert_failure_messages
+
+
+@pytest.mark.parametrize(
+    "arn, region, expected_service, expected_resource, expected_response",
+    [
+        (
+            "arn:aws:validate_service:us-west-2:123456789012:validate_resource",
+            "us-west-2",
+            "validate_service",
+            "validate_resource",
+            None,
+        ),
+        (
+            "arn:aws:s3:us-west-2:123456789012:bucket:my-bucket",
+            "us-west-2",
+            "validate_service",
+            "validate_resource",
+            "The arn:aws:s3:us-west-2:123456789012:bucket:my-bucket is not supported in region us-west-2.",
+        ),
+        (
+            "arn:aws:secretsmanager:us-west-2:123456789012:bucket:my-bucket",
+            "us-west-2",
+            "validate_service",
+            "validate_resource",
+            "The arn:aws:secretsmanager:us-west-2:123456789012:bucket:my-bucket "
+            "is not supported in region us-west-2.",
+        ),
+    ],
+)
+def test_arn_service_and_resource_validator(
+    arn,
+    region,
+    expected_service,
+    expected_resource,
+    expected_response,
+):
+    actual_response = ArnServiceAndResourceValidator().execute(
+        arn,
+        region,
+        expected_service,
+        expected_resource,
+    )
+    assert_failure_messages(actual_response, expected_response)
+
+
+@pytest.mark.parametrize(
+    "munge_key_secret_arn, mock_response, expected_response, error_from_aws_service, expected_failure_level",
+    [
+        (
+            "arn:aws:secretsmanager:us-west-2:123456789012:secret:testSecret",
+            # In Base64 encoding:
+            # - Every 3 bytes of input are represented as 4 characters in the encoded string.
+            # - The length of the encoded string must be a multiple of 4.
+            # - Valid Base64 characters include upper/lowercase letters,
+            #   numbers, '+', '/', and possibly trailing '=' padding chars.
+            # - The '=' padding is used if the number of bytes being encoded is not divisible by 3.
+            # Given these rules:
+            # - "validBase64" is valid because its length is a multiple of 4 and uses valid Base64 characters.
+            # - "invalidBase64" is invalid because its length is not a multiple of 4.
```

Comment:
This is a BRILLIANT comment. Thanks for this!

Reply:
Thank you Jacopo!